### ISSUE TITLE
Simplifying the cef/entities/bezier branches 

### ISSUE THREAD

**Issue by koraa** [inexor-game/code#124]
@IAmNotHanni, @aschaeffer, @a-teammate Please tell me, whether you are OK with these changes. If you are, I would soon do the split.

# Current status

At the moment we have have quite a few branches that deal with the cef/entities/bezier subsystems. These branches are:

* cef
* cef+entities
* cef+ipc+entities
* cef+ipc+entities+bezier
* cmake_update+cef+ipc

A quick research shows that `cef`, `cef+entities` and `cef+ipc+entities+bezier` are **obsolete**, because all of their commits are included within cef+ipc+entities, which is also based on a more recent version of the master.

`cmake_update+cef+ipc` integrates `cef+ipc+entities` and `cmake_update`. However it integrates older version of each, and it integrates them by merg cmake-update and thereby loses integration with master. It is therefore **obsolete**.

Since it's the only one left, `cef+ipc+entities` can now be considered **upstream** for the cef, bezier and entities subsystems!

# How to proceed

I suggest that we first delete the obsolete branches:

* cef
* cef+entities
* cef+ipc+entities+bezier
* cmake_update+cef+ipc

**NOT: cef+ipc+entities**

In order to allow a better integration, we should split the branch `cef+ipc+entities` into two:
`CefWebUI` and `NewEntities`. The Bezier subsystem can not be in it's own branch, since it is very closely tied to the entities feature.
This split could be done easily, since the first half of `cef+ipc+entities` contains exclusively changes for cef and the second half contains only commits for entities.

Of course, after splitting `cef+ipc+entities`, both resulting branches need to be rebased onto the current master.

## Integrating CEF

Since CEF is the foundation for many future features, I suggest that we get it into master as quickly as possible:
After it has been rebased, I suggest we adapt it to be opt-in: The old gui should still work, while the new one can be switched on via command line if the user desires it.
This allows developers to start working on the UI without being bothered to compile inexor themselves.

We might also move to src/ui/web folder to the root, as this would match the naming scheme for node/ and #107.

**Comment by koraa** [comment]
@aschaeffer By the way, what does aschaeffer/particles contain?

**Comment by aschaeffer** [comment]
You cannot just split cef+ipc+entities into 2 branches (CefWebUI, NewEntities) because the entities needs the reference pointers provided by CEF.

I agree to merge CEF into master quickly, BUT it has to work on the windows plattform first. As long as this isn't ensured I veto for merging.

Also, two GUIs doesn't make sense. Instead we should focus on implementing all essential UI functionalities on top of CEF quickly.

aschaeffer/particles is a completely seperated implementation of the particle system. The development on this branch is stopped, but I still need it for reference until I have reimplemented the complete functionality.

**Comment by koraa** [comment]
I now have created branches for CEF and for Entities. Since, however, entities can not be separated the entities branch is now based on CEF.
The branches are:
* CefWebUI
* NewEntities

@aschaeffer @a-teammate Can the now redundant branches be deleted? I am referring to: 
* cef
* cef+entities
* cef+ipc+entities
* cef+ipc+entities+bezier
* cmake_update+cef+ipc
* aschaeffer/particles

@aschaeffer What do you think of my suggestion to move the web UI fi

**Comment by IAmNotHanni** [comment]
bézier curves are a small module. They are not used / referenced in the code (yet)

**Comment by a-teammate** [comment]
for me its fine to delete cmake_update+cef+ipc idk about the rest

**Comment by Croydon** [comment]
We still have 5 CEF branches, are they still active?

**Comment by aschaeffer** [comment]
Only the pure CEF part is merged into master. The other branches contain the entity system, the particle system and the bezier curves which are not merged. I hate this issue. In fact there is no single reason for deleting these branches.

**Comment by Croydon** [comment]
@Hanack It's still hard to keep track of these branches or understand them.

Does one of these branches: 

> cef+entities
cef+ipc+entities
cef+ipc+entities+bezier
fix_cef
fix_cef_2704
rebased-cef+entities 

still contain changes/improvements _**- for CEF or IPC -**_ which aren't already merged in master?

**Comment by aschaeffer** [comment]
No, the CEF and IPC parts are already merged. The entity system not though.
